c++/src/kj/compat/http.c++

```diff
@@ -5418,7 +5418,20 @@ public:
   }
 
   kj::Promise<ConnectResponse> connect(kj::StringPtr host, const HttpHeaders& headers) override {
-    KJ_UNIMPLEMENTED("connect() is not implemented for NetworkHttpClient");
+    // We want to connect directly instead of going through a proxy here.
+    // https://github.com/capnproto/capnproto/pull/1454#discussion_r900414879
+    return network.parseAddress(host)
+        .then([this](auto address) {
+      return address->connect().then([this](auto connection) {
+        auto emptyHeaders = kj::heap<kj::HttpHeaders>(responseHeaderTable);
+        return ConnectResponse {
+          200, // .statusCode
+          "OK", // .statusText
+          emptyHeaders, // .headers
```

Comment:
You are using `emptyHeaders` here, but on the next line -- which is part of the same expression -- you are moving `emptyHeaders` away. Does C++ guarantee that these actions will execute in the order you expect? I'm not sure. I would normally copy a pointer to a local variable in the line before, to avoid the ambiguity.

Reply:
Just for future reference: This was resolved as part of the pipelining PR.